common/persistence/sql/sqlplugin/mysql/task.go

```diff
@@ -46,9 +46,23 @@ name = :name AND
 task_type = :task_type
 `
 
+	// This query uses pagination that is best understood by analogy to simple numbers.
+	// Given a list of numbers
+	// 	111
+	//	113
+	//	122
+	//	211
+	// where the hundreds digit corresponds to domain_id, the tens digit
+	// corresponds to name, and the ones digit corresponds to task_type,
+	// Imagine recurring queries with a limit of 1.
+	// For the second query to skip the first result and return 112, it must allow equal values in hundreds & tens, but it's OK because the ones digit is higher.
+	// For the third query, the ones digit is now lower but that's irrelevant because the tens digit is greater.
+	// For the fourth query, both the tens digit and ones digit are now lower but that's again irrelevant because now the hundreds digit is higher.
+	// This technique is useful since the size of the table can easily change between calls, making SKIP an unreliable method, while other db-specific things like rowids are not portable
```

Comment:
nit: just to clarify is it potential bug fix and performance optimization here?

Reply:
I think it's a bug fix. The original query will skip records during pagination.

Reply:
Yes, I would only describe this as a bug fix. I'm not sure it affects performance.

In the prior implementation, to draw from the analogy in the comment, only the first row in the example would be returned. None of the remaining three rows have all of their elements greater than the first, which is what it previously required. This mean the query skipped rather broadly through the dataset.

Reply:
I see, thanks much for the great contribution. We really, really appreciate it!